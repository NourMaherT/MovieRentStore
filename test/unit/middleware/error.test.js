const err = require('../../../middleware/error')

describe('Error middleware', () => {
    const scenarios = [
        {
          description: 'should return error 500 if somthing failed',
          error: 'Somthing Failed.',
          status: 500
        }
      ];
    scenarios.forEach(({ description, error, status }) => {
        it(description, async () => {
            const error = {
                message: 'Error message'
            };
            const req = {};
            const res = {
                send: jest.fn(),
                status: jest.fn(() => res)
              };
            const next = jest.fn();
    
            await err(error, req, res, next);
            expect(res.status).toBeCalledWith(status);
            expect(res.send.mock.calls[0][0]).toMatchSnapshot();
        });
    });

});
