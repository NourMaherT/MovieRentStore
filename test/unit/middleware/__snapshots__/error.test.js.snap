// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error middleware should return error 500 if somthing failed 1`] = `"Somthing Failed."`;
